generators/markup/files/render.js

```diff
@@ -0,0 +1,47 @@
+
+// React.
+import { createElement } from 'react';
+import { renderToStaticMarkup, renderToString } from 'react-dom/server';
+
+// Prevent XSS.
+import escape from 'htmlescape';
+
+// Generic page layout.
+import Page from './page';
+import Root from '../';
+
+function body(root, { scripts = [] }) {
+  // TODO: This is where any flux magic would go.
+  return new Promise((resolve) => {
```

Comment:
@jjt: @nealgranger basically has it. The file itself is the page rendering code. Given a root component, eventually render the entire page to a string.

The waiting part comes from smart components that trigger actions when they're instantiated. e.g. You could imagine a `Cart` smart component that fetches the current cart; this action is not triggered by the user, but instead triggered by the page itself.

 * Wait for current list of pending actions to resolve (initially empty).
 * Dehydrate state.
 * Render to string.
 * Dispatch actions generated from render call.
 * Update the list of pending actions.
 * If list is empty, return generated string.